newsroom/wire/search.py

```diff
@@ -518,3 +525,59 @@ def get_matching_bookmarks(self, item_ids, active_users, active_companies):
                     bookmark_users.append(bookmark)
 
         return bookmark_users
+
+    def permission_embeds_in_item(self, item, permitted_products):
```

Comment:
Could this be purely done in the browser instead? I'm wondering about the performance cost here, and what are the pros/cons of doing this server side or browser side